Remove distro-specific instructions and only leave links (#194)
haskell-infra/www.haskell.org: site/downloads.markdown

This page describes the installation of the Haskell toolchain, which consists of the following tools:

*   [GHC](https://www.haskell.org/ghc/): the Glasgow Haskell Compiler
*   [cabal-install](https://cabal.readthedocs.io): the Cabal installation tool for managing Haskell software
*   [Stack](https://docs.haskellstack.org): a cross-platform program for developing Haskell projects
*   [haskell-language-server](https://github.com/haskell/haskell-language-server) (optional): A language server for developers to integrate with their editor/IDE

> ## Recommended installation instructions

Reviewer: I wonder if we might not want to add something like `(All Operating Systems)` at the end here so someone glancing at the document will start here. I guess perhaps the `recommended` covers it, but I'd like to draw eyes to this section rather than the distro packages, since I think on average people will have a better experience avoiding the distro packages.

> *for Linux, macOS, FreeBSD, Windows or WSL2*

Reviewer: Nit-pick, but I wonder if we might want to remove this line altogether in light of the other changes?

1. Install GHC, cabal-install and haskell-language-server via [GHCup](https://www.haskell.org/ghcup/)

> 2. To install Stack, follow the [Stack installation guide](https://docs.haskellstack.org/en/stable/install_and_upgrade/)

Reviewer: Would it make sense for these to be bullet points rather than a numbered list? The numbered list implies to me that you need to follow each step in order, rather than having a choice of install method.

Author: Yes, that's a good point. However, the numbered list predates this PR so I suggest we leave that for a separate PR.

* * *

### Via native OS package manager

> Alternatively, many operating systems provide GHC, cabal and Stack
> through their native package manager.  The packages are often
> out-of-date but if you prefer to use this method of installation then
> you will find useful links below.

Reviewer 2: I'd lean towards a slightly different tone here:

```suggestion
Many operating systems provide GHC, cabal and Stack through their package managers. Note that OS packages can often be out of date.
```

#### Debian

* Official packages:
  [GHC](https://packages.debian.org/search?keywords=ghc&searchon=names&suite=all&section=all),
  [cabal](https://packages.debian.org/search?suite=all&section=all&arch=any&searchon=names&keywords=cabal-install),
  [Stack](https://packages.debian.org/search?suite=all&section=all&arch=any&searchon=names&keywords=haskell-stack)

#### Fedora

* Official packages
  [GHC](https://packages.fedoraproject.org/pkgs/ghc/ghc/),
  [cabal](https://packages.fedoraproject.org/pkgs/cabal-install/cabal-install/),
  [Stack](https://packages.fedoraproject.org/pkgs/haskell-platform/stack/)

#### openSUSE

* Official packages
  [GHC](https://software.opensuse.org/package/ghc?search_term=%22ghc%22),
  [cabal](https://software.opensuse.org/package/cabal-install),
* [devel:languages:haskell repo](https://build.opensuse.org/project/show/devel:languages:haskell)

#### Gentoo

* Official packages
  [GHC](https://packages.gentoo.org/packages/dev-lang/ghc),
  [cabal](https://packages.gentoo.org/packages/dev-haskell/cabal),
  [Stack](https://packages.gentoo.org/packages/dev-haskell/stack)
* [Third party repo](https://github.com/gentoo-haskell/gentoo-haskell)

#### Arch

> **Do not use the Haskell development tools provided by Arch, they are
> broken.** For more information see

Reviewer 2: ```suggestion
**Haskell tools and libraries natively packaged by Arch do not work for local Haskell development.** For more information see
```

Reviewer 2: Seems like we could be a bit more specific here—if I'm understanding the situation correctly, the Arch approach is fine for stuff like packaging Pandoc for Arch users, but it does not work well for *doing Haskell development in general*.

Author: Sure, but someone packaging Pandoc for Arch users already knows that, right?  The ratio among visitors to this page of  Arch users who are not trying to package for Arch : Arch developers who are trying to package for arch must be something like 1000 : 1, or more right?  In which case I think it's probably unhelpful to be more nuanced, even if it is technically correct.

Put another way, the people who would be negatively effected by the lack of nuance in the original wording already know the solution anyway, so I don't think there's much point going out of our way to accommodate them at the expense of presenting a clear message.

Reviewer 3: We want to make *very* clear to users that they shouldn't use arch packages. The word **broken** is key here.

Reviewer 2: Seems fine either way.
